ampd/src/queue/queued_broadcaster.rs

```diff
@@ -25,36 +29,36 @@ pub enum Error {
     Queue,
 }
 
-pub struct QueuedBroadcasterDriver {
-    #[allow(dead_code)]
-    broadcast_tx: mpsc::Sender<()>,
-}
-
-impl QueuedBroadcasterDriver {
-    #[allow(dead_code)]
-    pub async fn force_broadcast(&self) -> Result {
-        self.broadcast_tx
-            .send(())
-            .await
-            .map_err(|_| Report::new(Error::Broadcast))
-    }
-}
-
 #[automock]
 #[async_trait]
 pub trait BroadcasterClient {
     async fn broadcast(&self, tx: Any) -> Result;
 }
 
-pub struct QueuedBroadcasterClient {
-    sender: mpsc::Sender<Any>,
+pub struct QueuedBroadcasterClient<F>
+where
+    F: FeeEstimator,
+{
+    sender: mpsc::Sender<MsgAndFee>,
+    fee_estimator: Arc<Mutex<F>>,
 }
 
 #[async_trait]
-impl BroadcasterClient for QueuedBroadcasterClient {
+impl<F> BroadcasterClient for QueuedBroadcasterClient<F>
+where
+    F: FeeEstimator + Send,
+{
     async fn broadcast(&self, msg: Any) -> Result {
+        let fee = self
+            .fee_estimator
+            .lock()
```

Comment:
Why do we need to lock here? Also, is this going to contend with the lock acquired when broadcasting? Do we want to use a RwLock?

Reply:
Both fee_estimate and broadcast methods require a mutable reference of the broadcaster. Both of them update the sequence number maintained in memory. Even if they don't, they still both require a mutable reference of the underlying gRPC client, which is just the nature of gRPC, i.e. sending any request requires a mutable reference of the client. Therefore RwLock doesn't make sense.